src/doc/telio_events.md

```diff
@@ -0,0 +1,150 @@
+# Documentation For Describing Telio Events
+
+This module documents the events emitted by the Telio library.
+
+## Telio Error Event
+
+Once registered using telio_new for events. One of the possible reported event is of type [Error][telio_model::event::Error], in format:
+
+```json
+{
+  "type": "error",
+  "body": {
+    "level": "<critical|severe|warning|notice>",
+    "code": "<noerror|unknown>",
+    "msg": string?
+  }
+}
+```
+
+Fields:
+
+- level - error severity:
+
+    - critical - telio panicked
+    - severe - telio can continue operating, but parts of it can malfunction
+    - warning - warning that things can get wrong
+    - notice - not a critical error
+
+- code - common error code:
+
+    - noerror - default value, not an error.
+    - unknown - unknown error
+
+- msg - custom error string, can be empty
+
+
+### When error event is reported?
+
+One of the situations, when error event will be reported, is when panic happens on telio side.
+
+
+### Handling error events
+
+- level:
+
+    - critical -  telio panicked, so it must be restarted calling telio_destroy_hard() and starting init procedure from scratch (invoking telio_new() or similar path). Note, that destroying and re-creating telio should not be pe3rformed while handling incoming event (not on telio thread’s context).
+
+## Telio Node Event
+
+Once registered using telio_new for events. One of the possible reported event is of type [Node][telio_model::mesh::Node], in format:
+
+```json
+{
+  "type": "node",
+  "body": {
+    "identifier": "<Node UUID>",
+    "public_key": "<base64 key>",
+    "state": "<disconnected|connecting|connected>",
+    "link_state": "<null|down|up>"
+    "is_exit": bool,
+    "is_vpn": bool,
+    "ip_addresses": [ "meshnet ip" ] // VPN nodes have reserved addresses: ["10.5.0.1", "100.64.0.1"]
+    "allowed_ips": [ "<CIDR>" ],
+    "endpoint":"<ip:port>",
+    "hostname":"<Node hostname>",
+    "path": "<direct|relay>",
+  }
+}
+```
+
+
+Fields:
```

Comment:
nit: Since we already document fields of the `Node` structure and link to it here, I'm not sure we need to duplicate that here.